server/index.ts

```diff
@@ -47,125 +56,255 @@ app.get('/auth/login', (_req, res) => {
+      method: 'post',
+      url: 'https://accounts.spotify.com/api/token',
+      data: formData,
+      headers: {
+        Authorization: `Basic ${Buffer.from(
+          `${spotifyClientId}:${spotifyClientSecret}`,
+        ).toString('base64')}`,
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    });
+    accessToken = response.data.access_token;
+    expiresIn = response.data.expires_in;
+    refreshToken = response.data.refresh_token;
+    res.redirect('http://localhost:3000');
+  } catch (err) {
+    console.log(err);
+    res.send(err);
+  }
 });
 
-// token endpoint
+// token endpoint -- send server globals to front-end
 app.get('/auth/token', (_req, res) => {
-  res.json({ access_token: accessToken });
+  const tokenObj = {
+    access_token: accessToken,
+    expires_in: expiresIn,
+    refresh_token: refreshToken,
+  };
+  res.json(tokenObj);
 });
 
-// userInfo endpoint
-app.get('/auth/me', (_req, res) => {
-  const authMeOptions = {
-    url: 'https://api.spotify.com/v1/me',
-    headers: {
-      Authorization: `Bearer ${accessToken}`,
-      'Content-Type': 'application/json',
-    },
+// refresh token endpoint -- call endpoint to refresh server globals and send to front-end
+app.get('/auth/refresh_token/:rTok', async (req, res) => {
+  const { rTok } = req.params;
+  const formObj = {
+    grant_type: 'refresh_token',
+    refresh_token: rTok,
   };
+  const formData = new URLSearchParams(formObj).toString();
 
-  request.get(authMeOptions, function (error, response, body) {
-    if (!error && response.statusCode === 200) {
-      res.json(response.body);
-    } else {
-      console.log(
-        error,
-        response.body.error,
-        response.body.error.error_description,
-      );
-      res.end(body);
-    }
-  });
+  try {
+    const response = await axios({
+      method: 'post',
+      url: 'https://accounts.spotify.com/api/token',
+      data: formData,
+      headers: {
+        Authorization: `Basic ${Buffer.from(
+          `${spotifyClientId}:${spotifyClientSecret}`,
+        ).toString('base64')}`,
+      },
+    });
+    accessToken = response.data.access_token;
+    expiresIn = response.data.expires_in;
+    res.send({
+      access_token: accessToken,
+      expires_in: expiresIn,
+    });
+  } catch (err) {
+    console.log(err);
+    res.send(err);
+  }
 });
 
-// playlist endpoint
-app.get('/auth/me/playlist', (_req, res) => {
-  const authPlaylistOptions = {
-    url: 'https://api.spotify.com/v1/me/playlists',
-    headers: {
-      Authorization: `Bearer ${accessToken}`,
-      'Content-Type': 'application/json',
-    },
-  };
+// userInfo endpoint -- get user data
+app.get('/auth/me', async (_req, res) => {
+  try {
+    await axios
+      .get('https://api.spotify.com/v1/me', {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+          'Content-Type': 'application/json',
+        },
+      })
+      .then((response) => {
```

Comment:
While it's not wrong to use `.then()` blocks with `await`ed promises, it's rarely necessary. I'd recommend switching this and similar blocks over to drop the `.then()`.... I'm okay if you want to take note of this, though, and address this and similar items in a refactor branch/PR down the line.

Remember also that when you do that, you can destructure the one piece of `response` you care about, `data`.